src/pybind/mgr/cephadm/services/osd.py

```diff
@@ -30,13 +30,15 @@ def create_from_spec(self, drive_group: DriveGroupSpec) -> str:
             if not cmd:
                 logger.debug("No data_devices, skipping DriveGroup: {}".format(drive_group.service_id))
                 continue
-            ret_msg = self.create(host, cmd,
-                                              replace_osd_ids=drive_group.osd_id_claims.get(host, []))
+            # env_vars = [f"CEPH_VOLUME_OSDSPEC_AFFINITY={drive_group.service_id}"]
+            # disable this until https://github.com/ceph/ceph/pull/34835 is merged
```

Comment:
Instead of reverting this, can we make this somehow optional?

Reply:
I thought about this as well, but I'm not sure how. The workability depends on the ceph-osd version (no specific version can be determined though)